plugins/backend/catalog-backend-module-okta/src/providers/OktaOrgEntityProvider.ts

```diff
@@ -168,59 +173,75 @@ export class OktaOrgEntityProvider extends OktaEntityProvider {
           logger: this.logger,
         });
 
-        await Promise.allSettled(
-          Object.entries(oktaGroups).map(async ([_, group]) => {
-            const members: string[] = [];
-            await group.listUsers().each(user => {
-              try {
-                const userName = this.userNamingStrategy(user);
-                if (userResources[userName]) {
-                  members.push(userName);
+        for (const chunkOfGroups of chunk(
+          Object.values(oktaGroups),
+          this.chunkSize,
+        )) {
+          const promises = await Promise.allSettled(
```

Comment:
These are awaited already so actual promise results, maybe a better naming for the array?

Reply:
Done. thanks.